### PR TITLE
fix time bounds in daily and monthly statistics

### LINKED ISSUE #398: time values after daily_statistics (or monthly_statistics)
The time bounds in daily_statistics (or monthly_statistics) need a correction. For example, regarding hourly ERA5 data, if daily_statistics is applied after CMORization, the daily time values will be 2001, 1, 1, 11, 30, 0. It should be corrected to 2001, 1, 1, 12, 0, 0.

### PR DESCRIPTION
**Pull request by SarahAlidoost**
Before you start, please read [CONTRIBUTING.md](https://github.com/ESMValGroup/ESMValTool/blob/version2_development/CONTRIBUTING.md).

**Tasks**

-   [x] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [ ] This pull request has a descriptive title that can be used in a changelog
-   [ ] Add unit tests
-   [ ] Public functions should have a numpy-style docstring so they appear properly in the [API documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/api/esmvalcore.html). For all other functions a one line docstring is sufficient.
-   [ ] If writing a new/modified preprocessor function, please update the [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/preprocessor.html)
-   [ ] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [x] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
-   [ ] Please use `yamllint` to check that your YAML files do not contain mistakes
-   [ ] If you make backward incompatible changes to the recipe format, make a new pull request in the [ESMValTool repository](https://github.com/ESMValGroup/ESMValTool) and add the link below

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes #398

### CODE REVIEW

File: esmvalcore/preprocessor/_time.py [hunk 1]
```diff
@@ -237,6 +245,14 @@ def monthly_statistics(cube, operator='mean'):
...
     cube = cube.aggregated_by(['month_number', 'year'], operator)
+
+    # Correct the time coordinate
+    cube.coord('time').points = cube.coord('time').units.date2num([
+        cell.point.replace(day=15, hour=12, minute=0, second=0, microsecond=0)
+        for cell in cube.coord('time').cells()])
+
+    cube.coord('time').bounds = None
+    cube.coord('time').guess_bounds()
     return cube
```
Comment (valeriupredoi) [inline esmvalcore/preprocessor/_time.py]: do we really need `second` and `microsecond`?

**Comment by valeriupredoi** [comment]
nice one! can you please add a note in the docstrings and in the documentation about this, it's useful for users to know what they get out of this :grin:

File: esmvalcore/preprocessor/_time.py (hunk 1 above)
Comment (SarahAlidoost) [reply esmvalcore/preprocessor/_time.py]: Thanks, I removed them. The iris function does not change the `second` and `microsecond`.

**Comment by SarahAlidoost** [comment]
> nice one! can you please add a note in the docstrings and in the documentation about this, it's useful for users to know what they get out of this

The notes are added.

**Comment by mattiarighi** [comment]
@SarahAlidoost do you want to update this given @bouweandela's comment above?

**Comment by bouweandela** [comment]
Closing this pull request for now. Let's first continue the discussion in the issue and then a new pull request can be opened if needed.

**Pull request closed without merging**